packages/elements/src/clock/index.ts

```diff
@@ -502,23 +609,30 @@ export class Clock extends ResponsiveElement {
   */
   private generateButtonsTemplate (): TemplateResult {
     return html`
-      <div part="increment-button" role="button" data-key="${UP}"></div>
-      <div part="decrement-button" role="button" data-key="${DOWN}"></div>
+      <div ${ref(this.upButtonRef)}
+           part="increment-button"
+           aria-hidden="true"
+           active></div>
+      <div ${ref(this.downButtonRef)}
+           part="decrement-button"
+           aria-hidden="true"
+           active></div>
     `;
   }
 
   /**
   * Get template of segment
-  * @param name segment's name
+  * @param segment segment
   * @param value segment's value
-  * @param shiftAmount amount to shift
   * @returns {TemplateResult} template
   */
-  private generateSegmentTemplate (name: string, value: number): TemplateResult {
+  private generateSegmentTemplate (segment: Segment, value: number): TemplateResult {
+    const isActive = this.interactive && (segment === this.activeSegment);
     return html`
-      <div part="segment ${name}${ifDefined(this.isSegmentShifted(name) ? ' shifted' : '')}" tabindex="${ifDefined(this.interactive ? '0' : undefined)}">
+      <div part="segment ${segment}${this.isSegmentShifted(segment) ? ' shifted' : ''}"
+           ?active=${isActive}>
         ${padNumber(value, 2)}
-        ${this.interactive ? this.generateButtonsTemplate() : undefined}
+        ${isActive ? this.generateButtonsTemplate() : undefined}
```

Comment:
This is still forcing the re-generation of the whole template. Instead of button always have `active` why don't you utilise active attribute and use CSS to toggle buttons ?

```suggestion
         <div ${ref(this.upButtonRef)}
            part="increment-button"
            aria-hidden="true"
            active=${ifDefined(isActive || undefined)}></div>
          <div ${ref(this.downButtonRef)}
            part="decrement-button"
            aria-hidden="true"
            active=${ifDefined(isActive || undefined)}></div>
```

Reply:
If we do this way, default clock will have hidden button. Do we really want this?

Reply:
What is the issue with hidden button ? I only see that it should be more performant.

Reply:
It's ok if you agree to have buttons in default clock (You can see what changes in snapshot). I've updated to remove `generateButtonTemplate` and do your suggestion. But I add more condition by `${isActive ? ref(this.upButtonRef) : undefined}`.

Reply:
Please take a look at this PR https://github.com/Refinitiv/refinitiv-ui/pull/329, this is an alternative solution without using ref. Choose either this or multiple `ref` but not adding and removing `ref` on active change.